### PR TITLE
[Media] Ensure date of administration is not in the future

### LINKED ISSUE #4573: [media] date taken in the future
when uploading a media file, I can specify a date in the future and it will work without notice.
Same apply when editing a  previously uploaded file, it is possible to change the date taken to a future date

### PR DESCRIPTION
**Pull request by driusan**
Add check to verify that the date of administration is not
in the future, if provided.

Fixes #4573.

### CODE REVIEW

**Comment by driusan** [comment]
Blocked by #4827

**Comment by driusan** [comment]
@cmadjar done

**Comment by driusan** [comment]
@cmadjar I think that was fixed in a different PR, I rebased it so it should be working now if you re-test.

File: modules/media/ajax/FileUpload.php [hunk 1]
```diff
@@ -390,3 +396,27 @@ function getFilesList()
...
+{
+    if (!empty($dateTaken)) {
+        $date = date_create_from_format("Y-m-d", $dateTaken);
+        if ($date === false) {
+            showMediaError("Invalid date: $dateTaken", 400);
+        }
+
+        $now  = new DateTime();
+        $diff = date_diff($date, $now)->format("%a");
+        if ($diff > 0) {
```
Comment (jesscall) [inline modules/media/ajax/FileUpload.php]: @driusan Shouldn't this be `if ($diff < 0)` ??
`date_diff` does `$now - $date` which should technically be positive if the `$date` is in the past...

**Pull request merged**